test/js/ez-browse.js

```diff
@@ -0,0 +1,95 @@
+        );
+    });
+
+    describe('properties', function () {
+        describe('`selectedLocationId`', function () {
+            it('should be set by the attribute', function () {
+                const value = '42';
+
+                element.setAttribute('selected-location-id', value);
+                assert.equal(element.selectedLocationId, value);
+            });
+        });
+    });
+
+    describe('content discover', function () {
+        function simulateClick(element) {
+            const click = new CustomEvent('click', {
+                bubbles: true,
+                cancelable: true,
+            });
+
+            element.dispatchEvent(click);
+            return click;
+        }
+
+        it('should dispatch contentDiscover with config and listeners on click', function () {
+            element.addEventListener('ez:contentDiscover', (e) => {
+                assert.equal(e.detail.config.title, 'Browse', 'should provid the title');
+                assert.equal(e.detail.config.startingLocationId, 42, 'should provid the startingLoactionId');
+                assert.equal(e.detail.config.confirmLabel, 'View this content', 'should provid the confirmLabel');
+                assert.isFunction(e.detail.listeners['ez:confirm'], 'should provid `ez:confirm` eventListener');
+                assert.isTrue(e.bubbles, 'event should bubble');
+            });
+            simulateClick(element);
+        });
+
+        it('should dispatch contentDiscover with an undefined `startingLocationId` if `selectedLocationId` is "false"', function () {
+            element.selectedLocationId = 'false';
+            element.addEventListener('ez:contentDiscover', (e) => {
+                assert.isUndefined(e.detail.config.startingLocationId);
+            });
+            simulateClick(element);
+        });
+    });
+
+    describe('browse to location', function () {
+        function simulateClick(element) {
+            const click = new CustomEvent('click', {
+                bubbles: true,
+                cancelable: true,
+            });
+
+            element.dispatchEvent(click);
+            return click;
+        }
+
+        it('should dispatches `ez:navigateTo` event with the uri', function () {
+            const url = 'u/r/l/42/43';
+            const confirmFakeEventFacade = {
+                detail: {
+                    selection: {
+                        location: {
+                            id: 42,
+                            contentInfo: {
+                                id: 43,
+                            },
+                            url: url,
+                        },
+                    },
+                },
+            };
+
+            element.addEventListener('ez:contentDiscover', (e) => {
+                e.detail.listeners['ez:confirm'](confirmFakeEventFacade);
+            });
+
+            element.addEventListener('ez:navigateTo', (e) => {
+                assert.equal(e.detail.url, url);
```

Comment:
you don't check that the event is really dispatched, if it's not the test won't fail
also here, you should add a failing message